compose/config/config.py

```diff
@@ -431,7 +438,15 @@ def resolve_volume_paths(volumes, working_dir=None):
 def resolve_volume_path(volume, working_dir):
     container_path, host_path = split_path_mapping(volume)
     container_path = os.path.expanduser(container_path)
+
     if host_path is not None:
+        if not any(host_path.startswith(c) for c in PATH_START_CHARS):
+            log.warn(
```

Comment:
Agreed - I've updated it so that it specifies the service and links to https://github.com/docker/docker/pull/14242.